test-harness/src/AthenaQueryHandler.ts

```diff
@@ -0,0 +1,122 @@
+import {setTimeout} from "timers/promises";
+import {convertAthenaResultsToListOfRecords} from "./utils/ConvertAthenaResultToListOfRecords";
+
+const POWERTOOLS_METRICS_NAMESPACE = process.env.POWERTOOLS_METRICS_NAMESPACE ?? Constants.BAV_METRICS_NAMESPACE;
+const POWERTOOLS_LOG_LEVEL = process.env.POWERTOOLS_LOG_LEVEL ?? Constants.DEBUG;
+const POWERTOOLS_SERVICE_NAME = process.env.POWERTOOLS_SERVICE_NAME ?? Constants.DEQUEUE_LOGGER_SVC_NAME;
+
+export const logger = new Logger({
+	logLevel: POWERTOOLS_LOG_LEVEL,
+	serviceName: POWERTOOLS_SERVICE_NAME,
+});
+
+const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+
+export const athenaClient = new AthenaClient({region: process.env.REGION});
+
+class AthenaQueryHandler implements LambdaInterface {
+
+	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
+	async handler(event: APIGatewayProxyEvent, context: any): Promise<APIGatewayProxyResult> {
+
+		logger.setPersistentLogAttributes({});
+		logger.addContext(context);
+		logger.info("Starting AthenaQueryHandler");
+
+		if (!event.queryStringParameters) {
+			logger.error("Missing Query String Parameters");
+			return {
+				statusCode: 500,
+				body: "ERROR",
+			};
+		}
+
+		const sqlString = "SELECT itemnumber FROM \"" + (process.env["GLUE_TABLE"] ?? "") + "\" WHERE timestamp >= ? AND cicname LIKE ? ORDER BY timestamp DESC";
+		const queryExecutionInput : StartQueryExecutionInput = {
+			QueryString: sqlString,
+			QueryExecutionContext: {
+				Database: process.env["GLUE_DATABASE"],
+				Catalog: 'AwsDataCatalog',
+			},
+			ExecutionParameters: [
+				event.queryStringParameters['min-timestamp'] ?? "1",
+				event.queryStringParameters['name-prefix'] + "%" ?? "%",
+			],
+			WorkGroup: process.env["ATHENA_WORKGROUP"],
+		}
+
+		logger.debug("Sending query", {input: queryExecutionInput})
+		const startQueryExecutionCommandOutput = await athenaClient.send(new StartQueryExecutionCommand(queryExecutionInput));
+		logger.debug("Query start response", {output: startQueryExecutionCommandOutput})
+		const getQueryExecutionInput : GetQueryExecutionInput = {
+			QueryExecutionId: startQueryExecutionCommandOutput.QueryExecutionId,
+		}
+
+		let queryState: QueryExecutionState | undefined;
+		let getQueryExecutionCommandOutput : GetQueryExecutionCommandOutput;
+		do {
+			await setTimeout(500); //wait for pollInterval before calling again.
+			const getQueryExecutionCommand = new GetQueryExecutionCommand(getQueryExecutionInput);
+			logger.debug("Get Query input", {input: getQueryExecutionInput});
+			getQueryExecutionCommandOutput = await athenaClient.send(getQueryExecutionCommand);
+			logger.debug("Get Query output", {output: getQueryExecutionCommandOutput});
+			queryState = getQueryExecutionCommandOutput.QueryExecution?.Status?.State;
+		} while (queryState === QueryExecutionState.QUEUED || queryState === QueryExecutionState.RUNNING)
```

Comment:
I'm in two minds about this: yes we should catch timeouts/hanging on the AWS call and infinite loops, but ultimately this will be handled by the lambda and APIGW timeouts anyway, and this is just a test harness function so the timeout handling can be quite abrupt. I've lowered the lambda timeout to 30secs, and the APIGW timeout is 29secs by default, so this is the maximum it can run.